rasa/shared/core/domain.py

```diff
@@ -1916,6 +1919,66 @@ def _collect_actions_which_explicitly_need_domain(
 
         return action_names
 
+    @staticmethod
+    def _collect_response_ids(
+        responses: Dict[Text, List[Dict[Text, Any]]]
+    ) -> Dict[Text, Set[Text]]:
+        """Collects all response ids.
+
+        Args:
+            responses: The responses to collect the ids from.
+
+        Returns:
+            A dictionary mapping the response names to the set of response ids.
+        """
+        response_ids = set()
+        response_ids_per_response: Dict[Text, Set[Text]] = {}
+        for response_name, response_variations in responses.items():
+            response_ids_for_response = (
+                Domain._collect_response_ids_for_response_variations(
+                    response_variations
+                )
+            )
+
+            already_present_response_ids = response_ids_for_response.intersection(
+                response_ids
+            )
+
+            if len(already_present_response_ids) > 0:
+                rasa.shared.utils.io.raise_warning(
+                    f"Duplicate response ids "
+                    f"'{already_present_response_ids}' "
+                    f"defined in domain."
+                )
```

Comment:
Nice one! I'd suggest adding a follow-up ticket to improve the validation checks ran by `rasa data validate` to catch this particular issue (here or in line 1976)  if you think this is out of scope for this ticket.

Reply:
I created a [ticket](https://rasahq.atlassian.net/browse/ATO-1182) for it.